### PR TITLE
Frontend With FrameWorks - Comparing Functional and Class Components

### LINKED ISSUE #1880: Frontend With Framework : Comparing Functional and Class Components
### Description

Need to add a Comparision of Functional and Class Components.Different methods in them to do a specific task.Synatx,which one is better.

Table form will be better.

### Domain

Frontend Dev React/Angular/Vue

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by nooobcoder**
<hr>

## Description 📜

A documentation to compare **Functional and Class Based Components in ReactJS**.
[Fixes #1880](https://github.com/girlscript/winter-of-contributing/issues/1880)

Choosing between the two components design style is very important. But, choosing functional components approach is a wise thing 🧠

<hr>

## Type of change 📝
- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊
- [x] Frontend Dev React/Angular/Vue

<hr>

## Checklist ✅

- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I'm GWOC'21 contributor

<hr>

#### Contributed by [Ankur Paul](https://www.github.com/nooobcoder)

### CODE REVIEW

**Review by riasenpai — APPROVED** [review]
Hey LGTM! And I like your way to explain things but let us wait for the approval of supervisors and if they suggest any changes you can do that too!

**Comment by nooobcoder** [comment]
@anandhu720 @riasenpai I was unaware of JSX based code formatting in Markdown, my recent commit fixes that. @anandhu720 Please review that change, (a very minified change though)

JSX based code formatting displays code with proper syntax highlighting and was a very crucial part of the documentation.

Thank you @riasenpai for appreciating my way of explaining in documentation, this shall make my contributions more strong in the future to GWoC! 

Looking forward to future milestones.

**Review by nooobcoder — COMMENTED** [review]
Fixed code snippets -> JSX

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md
Comment (the-pro) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md:234]: I dont think this code snippet would work

**Review by heet-kumar — CHANGES_REQUESTED** [review]
Also try to add Website Reference links from where you have referred give the credits to them also as it is mentioned into the server-announcement channel of the discord server. So add References.

**Review by heet-kumar — COMMENTED** [review]
@nooobcoder buddy I liked the way you explained this topic is quite damm good.

**Comment by nooobcoder** [comment]
@heet-kumar Thank you so much for taking time to review through the documentation. Although I have taken extreme care to hyperlink to official React docs, however I missed attaching references at the end of my document. Your comment has made me aware, and I greatly appreciate it to preserve equal rights to all and share credits amongst the open source community.

I would be committing a final commit with the references separately.

For the meantime @the-pro I have fixed the [issue of code block as mentioned](https://github.com/girlscript/winter-of-contributing/pull/2107#discussion_r714678467). Do review it and provide feedback for changes to be made.
Also, I have well noted about mentioning references in my documentation, which I would be doing very shortly.

Rest assured, everything is clean and would be ready for merging.

Happy Coding,
Thank You @nooobcoder

**Comment by nooobcoder** [comment]
@the-pro I have also added reference links for additional reading and sources. Please let me know any further amendments to be made.
Eager to be merged soon! 💥 

Good Night! 🌕

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md
Comment (the-pro) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md:232]: This definitely works but it seems like a hack to make this work. While learning react someone shouldnt come across this  kind of code. I have made a small change to this. Have a look at it: https://codesandbox.io/s/nifty-sea-e590x?file=/src/App.js

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md
Comment (the-pro) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md:256]: I cannot understand this. Is it supposed to be the functional component alternative of that. I was just wondering cause this does not render anything. Even if the render is same you should write it at least.

**Comment by the-pro** [comment]
@nooobcoder really nice job with the PR. Just get done with these changes and I will merge it

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md
Comment (nooobcoder) [reply Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md:272]: Yes, this is the functional component based approach of the above class component approach. Although I had said it, but was less emphasized (not being bold)

Now I have fixed it to highlight properly.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md
Comment (nooobcoder) [reply Frontend_Web_Development_React_Angular_Vue/React/Week 2/2.1_Comparing Functional and Class Components.md]: > This definitely works but it seems like a hack to make this work. While learning react someone shouldnt come across this kind of code. I have made a small change to this. Have a look at it: https://codesandbox.io/s/nifty-sea-e590x?file=/src/App.js

### REVIEW ACCEPTED

The [code snippet ](https://codesandbox.io/s/nifty-sea-e590x?file=/src/App.js) was carefully seen and seemed much better and easy to me (though being a bit lengthy) and thus I have decided to accept your suggestion and has been reflected finally.

Thank you,
for showing such deep concern into the issue, and showing great support while contributing.

**Review by heet-kumar — APPROVED** [review]
I think from my side its ready for merge, let us wait for the approval of supervisors and if they suggest any changes kindly do that too!

**Review by the-pro — APPROVED** [review]

**Comment by nooobcoder** [comment]
Thank you @the-pro for the successful merge!
Sweet 🎉

**Pull request merged**
